tests/templates.py

```diff
@@ -56,14 +57,30 @@
 
 LOG_FILE="../`basename "$0"`".log
 rm -f "$LOG_FILE"
+ARTIFACT_ROOT="../../refactor-diffs"
+mkdir -p "$ARTIFACT_ROOT"
+PROJECT_NAME="$(basename "$SCRIPT_DIR")"
+ARTIFACT_DIR="${ARTIFACT_ROOT}/${PROJECT_NAME}"
+rm -rf "$ARTIFACT_DIR"
+mkdir -p "$ARTIFACT_DIR"
 
 while IFS= read -r transform; do
     [[ -z "$transform" ]] && continue
+    DIFF_FILE="$ARTIFACT_DIR/${transform}.diff"
+
     c2rust-refactor \
         ${transform} \
         --cargo --lib \
-        --rewrite-mode inplace \
-        2>&1 | tee -a "$LOG_FILE"
+        --rewrite-mode diff \
+        >"$DIFF_FILE" \
+        2>>"$LOG_FILE"
+
+    if [[ -s "$DIFF_FILE" ]] && grep -q '^@@' "$DIFF_FILE"; then
+        echo "Saved diff for ${transform} at ${DIFF_FILE}" >>"$LOG_FILE"
+        patch -p1 --batch <"$DIFF_FILE" >>"$LOG_FILE" 2>&1
```

Comment:
Can you move these two lines out of the if now?  `patch` should work on an empty diff I believe.